recaf-ui/src/main/java/software/coley/recaf/services/config/factories/NumberComponentFactory.java

```diff
@@ -0,0 +1,37 @@
+package software.coley.recaf.services.config.factories;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import javafx.scene.Node;
+import javafx.scene.control.TextField;
+import software.coley.observables.Observable;
+import software.coley.recaf.config.ConfigContainer;
+import software.coley.recaf.config.ConfigValue;
+import software.coley.recaf.services.config.TypedConfigComponentFactory;
+@ApplicationScoped
+public class NumberComponentFactory extends TypedConfigComponentFactory<Integer> {
+
+    @Inject
+    protected NumberComponentFactory() {
+        super(false, Integer.class);
+    }
+
+    @Override
+    public Node create(ConfigContainer container, ConfigValue<Integer> value) {
+        Observable<Integer> observable = value.getObservable();
+        String translationKey = container.getScopedId(value);
+        TextField textField = new TextField();
+        textField.setText(Integer.toString(observable.getValue()));
+        textField.textProperty().addListener((observableValue, oldValue, newValue) -> {
+            try {
+                int newValueAsInt = Integer.parseInt(newValue);
+                observable.setValue(newValueAsInt);
+            } catch (NumberFormatException e) {
+                textField.setText(oldValue);
+            }
```

Comment:
I'll play with the UX later to see if this feels right when in use. May have some odd cases like starting to write a negative number, backspacing to an empty string, etc